Ansible/playbook.yml

```diff
@@ -0,0 +1,136 @@
+---
+- name: Security Hardening on Ubuntu
+  hosts: all
+  gather_facts: no
+  become: yes
+  vars:
+    ssh_public_keys:
+      - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCj/QBVIaRSbOcbBCL898xdh6OHr44bbeBweIna/GbBkbGNwiBaiWmbw5bO1aA47V2aXzyITzphTXlkyFLHMLDrDw3sjRbuALmvE3Bo4X9OQ6S65M5xuO/CYegXjyix6xEay1GlXXnOU1x9c7akZkrFRXdaSZNuuG3bhsNGpzhh9sXxXn9kaE0MOui3aDeO4zTdfXxYPcH3gjSzqnB80rANiq54BHwqUL8c7byrNlWvkcSf8ldckx2SuJ8I8qJCazkEsS928c0KjFVwxzir7oUrvRQFoa+wtckXqqXe9UX0DIwKfuevytELFQn/xrt4+Zog+B7WVKLSdh4EcZkwBzgUozKnSzqeGvWVjf+oBLoz4z83rwijND2UjiqkpTFEDuCxfqhGISp8VUBDLhrSVOXDj7OaFP+GUEifasMEE2bg1HKNm8eIu0Vkh1S4UA5BpkMB6ITQEs2+m5OfLvgyPPX1iMOgEV/EV1pxz4Qmw8denwL2SijvYlJlhHHKh8/WdIU= vagrant@master"  # Replace with actual SSH public key
+      - "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCydBZtnehII5hsGLcN7Do6XvBm6tG4n4M4s3HoKRifDIpKEvw+MgZ0nSqexYs6nrN5iBdL0Vf+c3Mv3AYUTFKhDl2JTu1FkDMv0Htkr7lPi4FWKEl4+10cykggBtgJIi3AknSVET8UrWvEAxTYBIqAqTHOa2dylAD2OQHKzB7A6TmRIzNz45uZWkYgKGQDT+ZXVJbrXRurWzhQ6moJe9ETacKwkZfOXzIVehkgE1wtxgB3GygVAarOKfptgRBHlxZCObbBnUSwUPE4ZcvmZ5iRluyqlZxxveMH3Zr9pwFzuLpl278vGDnVRctpAOb1q0e1pU6uRzNmSNDLf5yIEeDf1V2XGHSe8ojjwi+LA1JVkIQYQ0mFgk+F20aSREX1TPIVo8VckHezvqdDC8Uuml6dVemRIB0pdRjk11/disOXLzC7ESDz1ASw5oq/P7Ehi3+CyOhGpkILxf79nog3T3/D0TyefsIoFgZiUfpQsv4axkKJd5vIwKH+o2DpzOh8sWM= vagrant@master"
+
+    new_user: "matusik"
+    new_user_password: "matusik"
+
+  tasks:
+
+    #- name: Test connection
+    #  local_action: command ssh -q -o BatchMode=yes -o ConnectTimeout=3 {{ inventory_hostname }}
+    #  register: test_user
+    #  ignore_errors: true
+    #  changed_when: false
+
+    - name: Update and Upgrade apt packages
+      apt:
```

Comment:
A good practice is to utilize built-in modules as follows `ansible.builtin.apt`.

Also, ansible-lint will be happy then, see e.g. https://www.redhat.com/sysadmin/ansible-lint